msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/IBToolTaskTests.cs

```diff
@@ -0,0 +1,173 @@
+			}
+
+			foreach (var item in Directory.EnumerateFiles (projectDir, "*.storyboard", SearchOption.AllDirectories))
+				interfaceDefinitions.Add (new TaskItem (item));
+
+			foreach (var item in Directory.EnumerateFiles (projectDir, "*.xib", SearchOption.AllDirectories))
+				interfaceDefinitions.Add (new TaskItem (item));
+
+			return new IBTool {
+				AppManifest = new TaskItem (Path.Combine (projectDir, "Info.plist")),
+				InterfaceDefinitions = interfaceDefinitions.ToArray (),
+				IntermediateOutputPath = intermediateOutputPath,
+				BuildEngine = new TestEngine (),
+				ResourcePrefix = "Resources",
+				ProjectDir = projectDir,
+				SdkPlatform = platform,
+				SdkVersion = version.ToString (),
+				SdkUsrPath = usr,
+				SdkBinPath = bin,
+				SdkRoot = root,
+			};
+		}
+
+		[Test]
+		public void TestBasicIBToolFunctionality ()
+		{
+			var tmp = Path.GetTempPath ();
+
+			Directory.CreateDirectory (tmp);
+
+			try {
+				var ibtool = CreateIBToolTask (PlatformFramework.iOS, "../MyIBToolLinkTest", tmp);
+				var bundleResources = new HashSet<string> ();
+
+				Assert.IsTrue (ibtool.Execute (), "Execution of IBTool task failed.");
+
+				foreach (var bundleResource in ibtool.BundleResources) {
+					Assert.IsTrue (File.Exists (bundleResource.ItemSpec), "File does not exist: {0}", bundleResource.ItemSpec);
+					Assert.IsNotNullOrEmpty (bundleResource.GetMetadata ("LogicalName"), "The 'LogicalName' metadata must be set.");
+					Assert.IsNotNullOrEmpty (bundleResource.GetMetadata ("Optimize"), "The 'Optimize' metadata must be set.");
+
+					bundleResources.Add (bundleResource.GetMetadata ("LogicalName"));
+				}
+
+				string[] expected = { "LaunchScreen~ipad.nib/objects-8.0+.nib",
+					"LaunchScreen~ipad.nib/runtime.nib",
+					"LaunchScreen~iphone.nib/objects-8.0+.nib",
+					"LaunchScreen~iphone.nib/runtime.nib",
+					"Main.storyboardc/BYZ-38-t0r-view-8bC-Xf-vdC~ipad.nib/objects-8.0+.nib",
+					"Main.storyboardc/BYZ-38-t0r-view-8bC-Xf-vdC~ipad.nib/runtime.nib",
+					"Main.storyboardc/BYZ-38-t0r-view-8bC-Xf-vdC~iphone.nib/objects-8.0+.nib",
+					"Main.storyboardc/BYZ-38-t0r-view-8bC-Xf-vdC~iphone.nib/runtime.nib",
+					"Main.storyboardc/UIViewController-BYZ-38-t0r~ipad.nib/objects-8.0+.nib",
+					"Main.storyboardc/UIViewController-BYZ-38-t0r~ipad.nib/runtime.nib",
+					"Main.storyboardc/UIViewController-BYZ-38-t0r~iphone.nib/objects-8.0+.nib",
+					"Main.storyboardc/UIViewController-BYZ-38-t0r~iphone.nib/runtime.nib",
+					"Main~ipad.storyboardc/Info-8.0+.plist",
+					"Main~ipad.storyboardc/Info.plist",
+					"Main~iphone.storyboardc/Info-8.0+.plist",
+					"Main~iphone.storyboardc/Info.plist"
+				};
+
+				foreach (var bundleResource in expected)
+					Assert.IsTrue (bundleResources.Contains (bundleResource), "BundleResources should include '{0}'", bundleResource);
+
+				Assert.AreEqual (expected.Length, bundleResources.Count, "Unexpected number of BundleResources");
+			} finally {
+				Directory.Delete (tmp, true);
```

Comment:
You're deleting /tmp here (or whatever `Path.GetTempPath ()` returns), which is probably what's causing all the HarnessExceptions.